src/services/fileServices/MdPageServices/PageService.ts

```diff
@@ -0,0 +1,379 @@
+import _ from "lodash"
+import { ok, err, Result, ResultAsync, okAsync, errAsync } from "neverthrow"
+
+import UserSessionData from "@root/classes/UserSessionData"
+import { BaseIsomerError } from "@root/errors/BaseError"
+import EmptyStringError from "@root/errors/EmptyStringError"
+import MissingResourceRoomError from "@root/errors/MissingResourceRoomError"
+import { NotFoundError } from "@root/errors/NotFoundError"
+import { ResourceRoomDirectoryService } from "@root/services/directoryServices/ResourceRoomDirectoryService"
+import {
+  CollectionPageName,
+  ContactUsPageName,
+  HomepageName,
+  PageName,
+  ResourceCategoryPageName,
+  ResourceRoomName,
+  StagingPermalink,
+  SubcollectionPageName,
+  UnlinkedPageName,
+  PathInfo,
+  Homepage,
+  PageInfo,
+  ContactUsPage,
+  ResourcePageInfo,
+} from "@root/types/pages"
+import { Brand } from "@root/types/util"
+
+import { CollectionPageService } from "./CollectionPageService"
+import { ContactUsPageService } from "./ContactUsPageService"
+import { HomepagePageService } from "./HomepagePageService"
+import { ResourcePageService } from "./ResourcePageService"
+import { SubcollectionPageService } from "./SubcollectionPageService"
+import { UnlinkedPageService } from "./UnlinkedPageService"
+
+// NOTE: This handler retrieves data from github then parses it
+// and it should handle the errors appropriately.
+// We return `BaseIsomerError` as a stop gap measure but in the future,
+// this should return a sum type of the possible errors.
+// This is (at a glance), either a failure to read from github
+// or the parsing fails (we have a sentinel value of an empty string)
+const withError = (promise: Promise<PageInfo>) =>
+  ResultAsync.fromPromise(promise, () => new BaseIsomerError()).andThen(
+    (data) =>
+      // NOTE: This is a fail-safe check as `yaml.parse`
+      // doesn't guarantee existence.
+      data?.content?.frontMatter?.permalink
+        ? ok(data)
```

Comment:
Hey, am confused on whats going on in this method call. we only run this method iff there is "either a a failure to read from github or the parsing fails (we have a sentinel value of an empty string)"... and if this is the case, why are we resolving it to an ok with `ok(data)` here ah?

Reply:
think you've a misunderstanding here. the comment is saying that the possible errors to come from here is either a failure to read from github, or a parse error (key doesn't exist for eg).

if there's a failure to read from github, 
`ResultAsync.fromPromise(promise, () => new BaseIsomerError())` resolves this because the promise will reject. This will turn into `ErrAsync<never, BaseIsomerError>` for the type, which is essentially `errAsync(new BaseIsomerError())` for the actual value. 

the check of `data?.content?.frontMatter?.permalink` is, as the comment implies, a fail-safe check to ensure that the property we care about (`permalink`) exists, as `yaml.parse` returns `any` and we want to make some guarantees on the shape of the object. 

this method runs on the `andThen`, which is only ran in the success case (ie, the github api call is successful), so this subsequent check is just to ensure the property exists

Reply:
ok, this makes sense, but shall we name the method call to something like `parseYamlFile` rather than just `withError`?

Reply:
actually this method is meant to act as a error handler over the raw promise. i was thinking about matching on the type of error returned from github and turning it into an actual internal error rather than just returning `BaseIsomerError`, which is not useful. 

i'm fine w/ renaming it but bc of ^, idt `parseYamlFile` is the right choice (as it also doesn't do parsing)

Reply:
got it, shall we rename this to `errorHandler` then?

Reply:
sure! i called it `withErrorHandler` as it injects it into the promise (as per what's described above!)